GDJS/Runtime/AsyncTasksManager.ts

```diff
@@ -58,13 +58,65 @@ namespace gdjs {
    */
   export abstract class AsyncTask {
     /**
-     * Called every frame where the scene is active.
+     * Called every frame where the scene is active and the task still unresolved.
      * @param runtimeScene - The scene the task runs on.
      * @return True if the task is finished, false if it needs to continue running.
      */
     abstract update(runtimeScene: RuntimeScene): boolean;
   }
 
+  /**
+   * A task that requires the usage of an object.
+   *
+   * If the object comes to be deleted, the task will automatically
+   * get rid of the object reference and resolve to avoid memory leaks.
+   */
+  export abstract class ObjectBoundTask extends AsyncTask {
+    public readonly objectInstance: RuntimeObject;
+    private hasBeenDeleted = false;
+
+    constructor(objectInstance: RuntimeObject) {
+      super();
+      this.objectInstance = objectInstance;
+      objectInstance.registerDestroyCallback(() => {
```

Comment:
We have to call `unregisterDestroyCallback` at some place, otherwise we risk creating a lot of tasks that will resolve and still be in memory because retained by the callback (which is retaining `this`) (and we'll lose precious CPU cycles too).